src/Data/List/Base.agda

```diff
@@ -229,6 +229,12 @@ lookup : ∀ (xs : List A) → Fin (length xs) → A
 lookup (x ∷ xs) zero    = x
 lookup (x ∷ xs) (suc i) = lookup xs i
 
+lookupMaybe : List A → ℕ → Maybe A
+lookupMaybe = λ where
+  []       _       → nothing
+  (x ∷ _)  zero    → just x
+  (_ ∷ xs) (suc n) → lookupMaybe xs n
+
```

Comment:
So... how useful/necessary is this operation?
It's noticeable that you don't add any lemmas about it to `Data.List.Properties`?

Reply:
Actually, this is often useful when writing meta-programs, and there is no real need for lemmas since we do not typically prove our meta-programs correct :)